CHANGELOG.md

```diff
@@ -1,5 +1,101 @@
+
+    __Before:__<br>
+    The `KnapsackPro::Hooks::Queue.after_queue` hook is executed outside of the `after(:suite)` hook.
+
+    __After:__<br>
+    The `KnapsackPro::Hooks::Queue.after_queue` hook is executed __inside__ of the `after(:suite)` hook.
+
+* RSpec recommended changes in your project:
+  * Remove the following code if you use Queue Mode and the `rspec_junit_formatter` gem to generate JUnit XML or JSON reports. The `KnapsackPro::Hooks::Queue.after_subset_queue` hook is no longer needed to generate a valid XML/JSON report.
+
+    ```ruby
+    # REMOVE THE FOLLOWING CODE
+    # spec_helper.rb or rails_helper.rb
+    TMP_REPORT = "tmp/rspec_#{ENV['KNAPSACK_PRO_CI_NODE_INDEX']}.xml"
+    FINAL_REPORT = "tmp/final_rspec_#{ENV['KNAPSACK_PRO_CI_NODE_INDEX']}.xml"
+
+    KnapsackPro::Hooks::Queue.after_subset_queue do |queue_id, subset_queue_id|
+      if File.exist?(TMP_REPORT)
+        FileUtils.mv(TMP_REPORT, FINAL_REPORT)
+      end
+    end
+    ```
+
+    You can keep the Knapsack Pro command unchanged with the `--format RspecJunitFormatter --out tmp/rspec.xml` options.
+
+    ```bash
+    bundle exec rake "knapsack_pro:queue:rspec[--format documentation --format RspecJunitFormatter --out tmp/rspec.xml]"
+    ```
+
+    Reference in [docs](https://docs.knapsackpro.com/ruby/rspec/#formatters-rspec_junit_formatter-json).
+
+    See [docs](https://docs.knapsackpro.com/ruby/circleci/) if you use CircleCI.
+
+  * Replace the following if you are using Queue Mode and have a version of the `percy-capybara` gem that is older than version `4`:
+
+    Before:
+
+    ```ruby
+    KnapsackPro::Hooks::Queue.before_queue { |queue_id| Percy::Capybara.initialize_build }
+    KnapsackPro::Hooks::Queue.after_queue { |queue_id| Percy::Capybara.finalize_build }
+    ```
+
+    After:
+
+    ```ruby
+    # recommended
+    before(:suite) { Percy::Capybara.initialize_build }
+    after(:suite) { Percy::Capybara.finalize_build }
+    ```
+
+    Reference in [docs](https://docs.knapsackpro.com/ruby/hooks/#percy-capybara) for the latest Percy version.
+
+  * We are no longer modifying the default RSpec formatters in Queue Mode. The `KNAPSACK_PRO_MODIFY_DEFAULT_RSPEC_FORMATTERS` environment variable is removed. You can remove it from the CI config if you are using it.
+
+* RSpec improvements in Queue Mode:
+  * The processing of termination signals - HUP, INT, TERM, ABRT, QUIT, USR1, and USR2 - has been improved for increased speed. Now, the process will terminate promptly before the execution of the next top-level RSpec example group (defined by `describe` or `context` blocks) within a spec file.
+    * It uses `1` as a default exit code when the process is terminated.
+    * Or, it uses a custom exit code defined by the `--error-exit-code` option.
+
+  * Respect the `--error-exit-code` option. It overrides the exit code when there are errors outside the test examples that RSpec cannot handle. For example, a lack of memory or a termination signal.
```

Comment:
```suggestion
  * Respect the `--error-exit-code` option. It sets a custom exit code (instead of `1`) when RSpec fails outside an example (e.g. lack of memory, termination signal).
```